src/nvme/fabrics.c

```diff
@@ -1026,8 +1031,11 @@ nvme_ctrl_t nvmf_connect_disc_entry(nvme_host_t h,
 		return NULL;
 	}
 
-	if (e->treq & NVMF_TREQ_DISABLE_SQFLOW)
+	if (e->treq & NVMF_TREQ_DISABLE_SQFLOW &&
+	    nvmf_check_option(h->r, disable_sqflow))
 		c->cfg.disable_sqflow = true;
+	else
+		c->cfg.disable_sqflow = false;
```

Comment:
Explicitly setting it to `false` is probably not necessary because `c->cfg` is 0-initialized by `nvmf_default_config()` (called from `nvme_create_ctrl()` above)

Reply:
Could we simply write:
```
c->cfg.disable_sqflow = (e->treq & NVMF_TREQ_DISABLE_SQFLOW) && nvmf_check_option(h->r, disable_sqflow);
```

Reply:
`cfg` is passed into the fucntion, so we don't have any control on the settings. We need to explicitly set it.

I find the explict if/else thing simpler too read, but I don't care too much. So if you insist I update it.

Reply:
But `c->cfg` is not derived from `cfg` as far as I could tell?
I would agree that not duplicating `c->cfg.disable_sqflow` in each of the cases makes it a bit easier to follow the purpose of this code, but I don't feel strongly.

Reply:
You are right. I confused `c->cfg` with `cfg`. In this case we don't need to set it explicitly to false.

Reply:
Should be okay now :)